frontend/src/home/AddPlanPopper.tsx

```diff
@@ -121,52 +108,121 @@ interface Props {
 }
 
+interface AddPlanPopperFields {
+  readonly planName: string | null;
+  readonly catalogYear: number | null;
+  readonly major: Major | null;
+  readonly concentration: string | null;
+  readonly coopCycle: string | null;
+  readonly planOption: string | null;
+  readonly basePlan: string | null;
+}
+
+const INIT_FIELDS: AddPlanPopperFields = {
+  planName: null,
+  catalogYear: null,
+  major: null,
+  concentration: null,
+  coopCycle: null,
+  planOption: PLAN_OPTIONS.NEW_PLAN,
+  basePlan: null,
+};
+
+const validateValues = (
+  scheduleNames: string[],
+  selectedDNDSchedule: React.MutableRefObject<DNDSchedule | undefined>,
+  noConcentrationError: boolean
+) => (values: AddPlanPopperFields): FormErrors => {
+  const validatePlanName = (values: AddPlanPopperFields) => {
+    // Missing plan name
+    if (!values.planName) {
+      return REQUIRED_FIELD_MESSAGE;
+    }
+
+    // Duplicate plan name
+    if (!!values.planName && scheduleNames.includes(values.planName!)) {
+      return "Cannot have the same name as an existing plan";
+    }
+  };
+
+  const validatePlanOption = (values: AddPlanPopperFields) => {
+    // Missing plan option
+    if (!values.planOption) {
+      return REQUIRED_FIELD_MESSAGE;
+    }
+  };
+
+  const validateBasePlan = (values: AddPlanPopperFields) => {
+    // Missing base plan
+    if (
+      values.planOption === PLAN_OPTIONS.EXISTING_PLAN &&
+      (!values.basePlan || !selectedDNDSchedule.current)
+    ) {
+      return REQUIRED_FIELD_MESSAGE;
+    }
+  };
+
+  const validateConcentration = (values: AddPlanPopperFields) => {
+    // Missing concentration
+    if (noConcentrationError) {
+      return REQUIRED_FIELD_MESSAGE;
+    }
+  };
+
+  return {
+    planName: validatePlanName(values),
+    planOption: validatePlanOption(values),
+    basePlan: validateBasePlan(values),
+    concentration: validateConcentration(values),
+  };
+};
+
 /**
  * If a user is currently logged in, saves the current plan under this user.
  * Only supports updating a user's singular plan, can be modified later to
  * update a specific plan.
  */
-function AddPlanPopperComponent(props: Props) {
-  const {
-    allMajors,
-    allPlans,
-    isFetchingMajors,
-    isFetchingPlans,
-    userPlans,
-    userId,
-    addNewPlan,
-  } = props;
-  const [visible, setVisible] = useState(false);
-  const [planName, setPlanName] = useState<string | null>(null);
-  const [selectedCatalogYear, setSelectedCatalogYear] = useState<number | null>(
-    null
-  );
-  const [selectedMajor, setSelectedMajor] = useState<Major | null>(null);
-  const [selectedConcentration, setSelectedConcentration] = useState<
-    string | null
-  >(null);
-  const [selectedCoopCycle, setSelectedCoopCycle] = useState<string | null>(
-    null
-  );
-  const [selectedPlanOption, setSelectedPlanOption] = useState<string | null>(
-    null
-  );
-  const [selectedUserPlan, setSelectedUserPlan] = useState<string | null>(null);
```

Comment:
All this state now is within a single state object that is contained within the `useForm` hook.